onebusaway-api-webapp/src/main/java/org/onebusaway/api/actions/api/gtfs_realtime/TripUpdatesForAgencyAction.java

```diff
@@ -28,19 +29,25 @@
 import com.google.transit.realtime.GtfsRealtime.VehicleDescriptor;
 import org.onebusaway.util.AgencyAndIdLibrary;
 
+import javax.servlet.http.HttpServletResponse;
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
+import java.util.*;
 
 public class TripUpdatesForAgencyAction extends GtfsRealtimeActionSupport {
 
   private static final long serialVersionUID = 2L;
 
+  private static final SimpleDateFormat _sdf;
+
+  static {
+    _sdf = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss 'GMT' ", Locale.US);
+    _sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+  }
   @Override
   protected void fillFeedMessage(FeedMessage.Builder feed, String agencyId,
       long timestamp, FILTER_TYPE filterType, String filterValue) {
 
+    long feedTimestamp = feed.getHeader().getTimestamp();
```

Comment:
What if timestamp isn't present?  I believe you need to call hasTimestamp() to verify one is set first....

Reply:
Yes, It should be.will correct it to use:
`long feedTimestamp = feed.getHeader().hasTimestamp() ? feed.getHeader().getTimestamp() : timestamp;`